sdk/communication/Azure.Communication.Sms/src/SmsClient.cs

```diff
@@ -34,7 +34,7 @@ public SmsClient(Uri endpoint, AzureKeyCredential keyCredential, SmsClientOption
         /// <summary> Initializes a new instance of <see cref="SmsClient"/>.</summary>
         /// <param name="connectionString">Connection string acquired from the Azure Communication Services resource.</param>
         /// <param name="options">Client option exposing <see cref="ClientOptions.Diagnostics"/>, <see cref="ClientOptions.Retry"/>, <see cref="ClientOptions.Transport"/>, etc.</param>
-        public SmsClient(string connectionString, SmsClientOptions? options = default)
+        public SmsClient(string connectionString, SmsClientOptions options = default)
```

Comment:
let's split this constructor into two. One with only `connectionString` and one with both `connectionString` and non-optional `options` to follow the guidelines. The rest of constructors can stay as is. Only the basic constructor needs the simple format. https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-client